libvmi/driver/xen/xen_events_legacy.c

```diff
@@ -925,7 +947,7 @@ xen_set_mem_access_legacy(vmi_instance_t vmi, mem_access_event_t *event,
 
         hvmmem_access_t access = memaccess_conversion[page_access_flag];
 
-        rc = xen->libxcw.xc_hvm_set_mem_access(xch, dom, access, page_key, npages);
+        rc = xen->libxcw.xc_hvm_set_mem_access(xch, dom, access, gpfn, 1);
```

Comment:
Small suggestion:
Maybe you could use a define here (and in similar cases).
It may be a matter of personal taste, but I find
`rc = xen->libxcw.xc_hvm_set_mem_access(xch, dom, access, gpfn, SINGLE_PAGE);`
much easier to read, especially when the definition of the function only states " uint64_t nr" as the type / name (which is not very helpfull for people who are new to xen / libvmi).